packages/framework-core/src/services/graphql/websocket-protocol/graphql-websocket-protocol.ts

```diff
@@ -39,34 +47,48 @@ export class GraphQLWebsocketHandler {
 
     try {
       this.logger.debug('Handling websocket message')
+      if ('error' in envelope) {
+        throw envelope.error
+      }
       if (!envelope.value) {
         throw new Error('Received an empty GraphQL body')
       }
+
       const clientMessage = envelope.value as GraphQLClientMessage
       this.logger.debug('Received client message: ', clientMessage)
       switch (clientMessage.type) {
         case MessageTypes.GQL_CONNECTION_INIT:
-          return await this.handleInit(envelope.connectionID)
+          return await this.handleInit(envelope.connectionID, clientMessage)
         case MessageTypes.GQL_START:
           return await this.handleStart(envelope.connectionID, envelope, clientMessage)
         case MessageTypes.GQL_STOP:
           return await this.handleStop(envelope.connectionID, clientMessage)
         case MessageTypes.GQL_CONNECTION_TERMINATE:
           return await this.handleTerminate(envelope.connectionID)
-          break
         default:
           // This branch should be impossible, but just in case
           throw new Error(`Unknown message type. Message=${clientMessage}`)
       }
     } catch (e) {
       this.logger.error(e)
-      await this.sendToConnection(envelope.connectionID, new GraphQLInitError(e.message))
+      await this.connectionManager.sendMessage(this.config, envelope.connectionID, new GraphQLInitError(e.message))
     }
   }
 
-  private async handleInit(connectionID: string): Promise<void> {
+  private async handleInit(connectionID: string, clientMessage: GraphQLInit): Promise<void> {
+    let userEnvelope: UserEnvelope | undefined
```

Comment:
If its a `let` no need to specify `undefined`

```suggestion
    let userEnvelope: UserEnvelope
```

Reply:
If I don't say that `userEnvelope` can be `undefined`, then I won't be able to assign undefined to it and the code will not compile.
To my knowledge, `let` has nothing to do with the type of the variable, but with its scope and mutability

Reply:
Oh that's weird. In theory, at the type level `let` acts as an alias to a type guard that adds `undefined` to the type of the reference.

That is because when you define `let` at the first time, it will be `undefined`. BUT! If you are willing to assign `undefined` by hand it might not work I guess. Thanks!